File: CppPractice/CppLeetcode/Leetcode228.cpp
#include<vector>
#include<string>
using namespace std;
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) 
    {
        vector<string> ans;
        bool isConcatenate  = false;
        int start = 0;
        int end = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            if (!isConcatenate)
            {
                start = nums[i];
                if (i+1<nums.size() && nums[i+1] - nums[i]==1)
                {
                    isConcatenate = true;
                }
                else
                {
                    ans.push_back(to_string(start));
                    isConcatenate = false;
                }
            }
            else
            {
                end= nums[i];
                string tempAns = to_string(start)+"->"+to_string(end);
                if (i+1<nums.size() && nums[i+1] == nums[i]+1)
                {
                    isConcatenate = true;
                }
                else
                {
                    ans.push_back(tempAns);
                    isConcatenate = false;
                }
            }
        }
        return ans;
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode228.cs

public class Solution228 {
    public IList<string> SummaryRanges(int[] nums) {
        IList<string> ans = new List<string>();
        string tempString = "";
        int start = 0;
        int end = 0;


        bool isConcatenate = false;
        for(int i=0; i<nums.Length ; i++)
        {
            if(!isConcatenate)
            {
                tempString = nums[i].ToString();
                if(i+1<nums.Length && nums[i+1]-nums[i]==1)
                {
                    start = nums[i] ; 
                    end = nums[i];
                    isConcatenate =true;
                    tempString = start.ToString()+"->"+end.ToString();
                }
                else
                {
                    ans.Add(tempString);
                    isConcatenate = false;
                }
            }
            else
            {
                end = nums[i];
                tempString = start.ToString()+"->"+end.ToString();
                if(!(i+1<nums.Length && nums[i+1]-nums[i]==1))
                {

                    ans.Add(tempString);
                    isConcatenate =false;
                }
            }
        }
        return ans;
    }
}